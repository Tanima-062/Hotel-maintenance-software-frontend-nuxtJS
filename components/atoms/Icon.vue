<template>
  <div :class="styles">
    <svg
      class="w-7 h-7 fill-current"
      :class="`${color}`"
      :style="`width: ${iconWidth}px; height: ${iconHeight}px`"
      :viewBox="`0 0 24 24`"
    >
      <path :d="icon" />
    </svg>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  icon!: string

  @Prop({ default: 'text-white' })
  color!: string

  @Prop({ default: '' })
  bgColor!: string

  @Prop({ default: false })
  round!: boolean

  @Prop({ default: 24 })
  iconWidth!: number

  @Prop({ default: 24 })
  iconHeight!: number

  @Prop({ default: '' })
  extraClass!: string

  get styles(): string {
    const classList = []
    if (this.bgColor) {
      classList.push(`${this.bgColor}`)
    }
    if (this.round) {
      classList.push('rounded-full')
    }
    if (this.extraClass) {
      classList.push(this.extraClass)
    }
    return classList.join(' ')
  }
}
</script>
