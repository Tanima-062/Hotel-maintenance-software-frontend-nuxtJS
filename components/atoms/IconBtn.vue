<template>
  <button
    type="button"
    class="flex h-fit-content items-center justify-center"
    :class="styles"
    :disabled="disabled"
    @click="click"
  >
    <svg
      class="w-7 h-7 fill-current"
      :class="`${color} hover:${hoverColor}`"
      :style="`width: ${iconWidth}px; height: ${iconHeight}px`"
      viewBox="0 0 24 24"
    >
      <path :d="icon" />
    </svg>
  </button>
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  icon!: string

  @Prop({ default: 'text-white' })
  color!: string

  @Prop({ default: 'text-white' })
  hoverColor!: string

  @Prop({ default: false })
  disabled!: boolean

  @Prop({ default: false })
  round!: boolean

  @Prop({ default: '' })
  bgColor!: string

  @Prop({ default: '' })
  padding!: string

  @Prop({ default: 24 })
  iconWidth!: number

  @Prop({ default: 24 })
  iconHeight!: number

  @Prop({ default: '' })
  extraClass!: string

  private isAnim: boolean = false
  private interval: number | undefined

  get styles(): string {
    const classList = []

    classList.push(this.isAnim ? 'btn--click' : 'btn')
    if (this.disabled) {
      classList.push('cursor-not-allowed')
    }
    if (this.round) {
      classList.push('rounded-full')
    } else if (!this.extraClass.includes('rounded-sm')) {
      classList.push('rounded-md')
    }
    classList.push(this.bgColor ? `${this.bgColor} opacity-75` : 'opacity-85')
    if (this.padding) {
      classList.push(`${this.padding}`)
    }
    if (this.extraClass) {
      classList.push(this.extraClass)
    }
    return classList.join(' ')
  }

  private doAnim(): void {
    if (this.isAnim || this.interval) {
      return
    }
    this.isAnim = true
    this.interval = window.setInterval(() => {
      this.isAnim = false
      clearInterval(this.interval)
      this.interval = undefined
    }, 130)
  }

  @Emit()
  private click(e: Event): Event {
    this.doAnim()
    return e
  }
}
</script>
<style lang="sass" scoped>
.btn
  transition: opacity .1s ease-in
  &:hover
    transition: opacity .2s cubic-bezier(.4,0,.6,1)
    opacity: 1
  &--click
    transition: opacity .05s ease-out
    opacity: .7
.h-fit-content
  height: fit-content
</style>
