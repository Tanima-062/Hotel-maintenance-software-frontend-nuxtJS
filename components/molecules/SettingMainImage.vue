<template>
  <div
    class="rounded border-2 border-dashed border-bdColor flex items-center justify-center flex-col transition-all duration-200 ease-in-out"
    :class="styles"
    @dragover.prevent="isHover = true"
    @dragleave="isHover = false"
    @drop.stop="drop"
  >
    <Icon
      class="mb-1"
      :color="isHover ? 'text-white' : 'text-tagColor'"
      :icon="wallpaperIcon"
      :icon-width="33"
      :icon-height="33"
    />
    <p
      class="text-sm text-center"
      :class="isHover ? 'text-white' : 'text-black-light'"
    >
      {{ $t('album.mainImageSection.DropZoneFirstLine') }}<br />{{
        $t('album.mainImageSection.DropZoneSecondLine')
      }}
    </p>
  </div>
</template>
<script lang="ts">
import { Component, Prop, Vue, Emit } from 'nuxt-property-decorator'

import { mdiWallpaper } from '@mdi/js'
@Component({
  components: {
    IconBtn: () => import('~/components/atoms/IconBtn.vue'),
    Icon: () => import('~/components/atoms/Icon.vue'),
  },
})
export default class extends Vue {
  @Prop({ default: 'w-60' })
  width!: string

  @Prop({ default: 'h-40' })
  height!: string

  private wallpaperIcon: string = mdiWallpaper
  private isHover: boolean = false

  get styles(): string {
    const classList = []
    classList.push(this.width)
    classList.push(this.height)
    classList.push(this.isHover ? 'bg-bdColor' : '')
    return classList.join(' ')
  }

  @Emit()
  private drop(event: Event & { dataTransfer: DataTransfer }): number {
    this.isHover = false
    const imageId = Number(event.dataTransfer.getData('text/plain'))
    event.dataTransfer.clearData()
    if (imageId && !Number.isNaN(imageId)) {
      return imageId
    }
    return 0
  }
}
</script>
