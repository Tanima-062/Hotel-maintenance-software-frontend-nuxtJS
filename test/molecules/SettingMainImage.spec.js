import { mount } from '@vue/test-utils'
import Vue from 'vue'
import VueI18n from 'vue-i18n'
import SettingMainImage from '@/components/molecules/SettingMainImage.vue'
import Icon from '@/components/atoms/Icon.vue'
import IconBtn from '@/components/atoms/IconBtn.vue'

Vue.use(VueI18n)
describe('Molecules SettingMainImage.vue Check', () => {
  test('SettingMainImage.vue Test', () => {
    const wrapper = mount(SettingMainImage, {
      stubs: {
        Icon,
        IconBtn,
      },
      mocks: {
        $t: (msg) => msg,
      },
    })
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
