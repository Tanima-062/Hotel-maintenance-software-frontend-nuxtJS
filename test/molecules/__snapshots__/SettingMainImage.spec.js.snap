// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules SettingMainImage.vue Check SettingMainImage.vue Test 1`] = `
<div
  class="rounded border-2 border-dashed border-bdColor flex items-center justify-center flex-col transition-all duration-200 ease-in-out w-60 h-40 "
>
  <div
    class="mb-1"
  >
    <svg
      class="w-7 h-7 fill-current text-tagColor"
      style="width: 33px; height: 33px;"
      viewBox="0 0 24 24"
    >
      <path
        d="M4,4H11V2H4A2,2 0 0,0 2,4V11H4V4M10,13L6,18H18L15,14L12.97,16.71L10,13M17,8.5A1.5,1.5 0 0,0 15.5,7A1.5,1.5 0 0,0 14,8.5A1.5,1.5 0 0,0 15.5,10A1.5,1.5 0 0,0 17,8.5M20,2H13V4H20V11H22V4A2,2 0 0,0 20,2M20,20H13V22H20A2,2 0 0,0 22,20V13H20V20M4,13H2V20A2,2 0 0,0 4,22H11V20H4V13Z"
      />
    </svg>
  </div>
   
  <p
    class="text-sm text-center text-black-light"
  >
    
    album.mainImageSection.DropZoneFirstLine
    <br />
    album.mainImageSection.DropZoneSecondLine
  
  </p>
</div>
`;
